Assets/HoloToolkit/Input/README.md

```diff
@@ -3,50 +3,649 @@
+    {
+        base.OnSourceDetected(eventData);
+        Debug.LogFormat("Joystick {0} with id: \"{1}\" Connected", GamePadName, eventData.SourceId);
+    }
+
+    public override void OnSourceLost(SourceStateEventData eventData)
+    {
+        Debug.LogFormat("Joystick {0} with id: \"{1}\" Disconnected", GamePadName, eventData.SourceId);
+        base.OnSourceLost(eventData);
+    }
+
+    public override void OnXboxInputUpdate(XboxControllerEventData eventData)
+    {
+        // FYI, eventData.Use() is called in the base implementation.
+        base.OnXboxInputUpdate(eventData);
+
+        Debug.LogFormat(
+            "{19}\n" +
+            "LS Horizontal: {0:0.000} Vertical: {1:0.000}\n" +
+            "RS Horizontal: {2:0.000} Vertical: {3:0.000}\n" +
+            "DP Horizontal: {4:0.000} Vertical: {5:0.000}\n" +
+            "Left Trigger:  {6:0.000} Right Trigger: {7:0.000} Shared Trigger: {8:0.00}\n" +
+            "A: {9} B: {10} X: {11} Y: {12}\n" +
+            "LB: {13} RB: {14} " +
+            "LS: {15} RS: {16}\n" +
+            "View: {17} Menu: {18}\n",
+            eventData.XboxLeftStickHorizontalAxis, eventData.XboxLeftStickVerticalAxis,
+            eventData.XboxRightStickHorizontalAxis, eventData.XboxRightStickVerticalAxis,
+            eventData.XboxDpadHorizontalAxis, eventData.XboxDpadVerticalAxis,
+            eventData.XboxLeftTriggerAxis, eventData.XboxRightTriggerAxis, eventData.XboxSharedTriggerAxis,
+            eventData.XboxA_Pressed, eventData.XboxB_Pressed, eventData.XboxX_Pressed, eventData.XboxY_Pressed,
+            eventData.XboxLeftBumper_Pressed, eventData.XboxRightBumper_Pressed,
+            eventData.XboxLeftStick_Pressed, eventData.XboxRightStick_Pressed,
+            eventData.XboxView_Pressed, eventData.XboxMenu_Pressed,
+            GamePadName);
+    }
+}
+```
+
+### Voice Input
+
+> _Note: Only availible for Windows and UWP Platform Targets._
+
+1. Voice Input
+2. Create a new scene
+3. Run the MRTK scene wizard via `MixedRealityToolkit/Configure/Apply Scene Settings`
+4. Create a Cube
+5. Create a new script named `SpeechHandler`
+6. Attach the `SpeechHandler` to your Cube
+7. Open the new script in any text editor
+8. Implement the `ISpeechHandler` interface
+9. Add switch statement for `eventData.RecognizedText` with a case for each command you wish to use.
+    - _Note: It’s helpful to use `ToLower()` to get better matches._
+
+```
+using UnityEngine;
+using HoloToolkit.Unity.InputModule;
+
+public class SpeechHandler : MonoBehaviour, ISpeechHandler
+{
+    void ISpeechHandler.OnSpeechKeywordRecognized(SpeechEventData eventData)
+    {
+        switch (eventData.RecognizedText.ToLower())
+        {
+            case "your voice command":
+                DoAction();
+                break;
+        }
+    }
+
+    public void DoAction()
+    {
+        // TODO: Action
+    }
+}
+```
+
+### Dictation Input
+
+> _Note: Only availible for Windows and UWP Platform Targets._
```

Comment:
Add a note here that mentions that you cannot use voice and dictation at the same time.

Reply:
Will do

Reply:
It's already built into the system to prevent people from doing exactly this.

Reply:
I'm not sure if this is exactly where we want to add this note, but down below in the class overview